parsl/dataflow/dflow.py

```diff
@@ -717,3 +718,58 @@ def load_checkpoints(self, checkpointDirs):
             raise BadCheckpoint("checkpointDirs expects a list of checkpoints")
 
         return self._load_checkpoints(checkpointDirs)
+
+
+class DataFlowKernelLoader(object):
+    """Manage which DataFlowKernel is active.
+
+    This is a singleton class containing only class methods. You should not need to instantiate this class.
+    """
+
+    _dfk = None
+
+    @classmethod
+    def set_default(cls, payload, **kwargs):
+        """If there is no active DataFlowKernel, load one.
+
+        Args:
+            - payload (str or DataFlowKernel) : Payload to load in case there is no active DataFlowKernel.
+              If a string is passed, it should be the path to a file which defines a dictionary named `config`;
+              this will be passed to the DataFlowKernel constructor.
+            - **kwargs : Arbitrary keyword arguments which are passed to the DataFlowKernel constructor.
+
+        """
+        if cls._dfk is None:
+            cls.load(payload, **kwargs)
+
+    @classmethod
+    def load(cls, payload, **kwargs):
+        """Load a DataFlowKernel.
+
+        Args:
+            - payload (str or DataFlowKernel) : Payload to load. If a string is passed, it should be the path to
+              a file which defines a dictionary named `config`. This config will be passed to a new DataFlowKernel
+              instantiation which will be set as the active DataFlowKernel. If a DataFlowKernel instance is passed,
+              it will be set as the active DataFlowKernel.
+            - **kwargs : Arbitrary keyword arguments which are passed to the DataFlowKernel constructor.
+        Returns:
+            - DataFlowKernel : The loaded DataFlowKernel object.
+        """
+        if isinstance(payload, DataFlowKernel):
+            cls._dfk = payload
+        else:
+            spec = importlib.util.spec_from_file_location("", payload)
+            module = importlib.util.module_from_spec(spec)
+            spec.loader.exec_module(module)
+            if hasattr(module, 'config'):
+                cls._dfk = DataFlowKernel(config=module.config, **kwargs)
```

Comment:
I don't think this is a good/clean idea.  Why would we not allow the payload to be a config ? I might be missing some advantage to this method.

Reply:
> Why would we not allow the payload to be a config ?

To make sure I understand, are you asking, "why would we allow the payload to be anything other than a config"? The main motivation was not to force users to always have the config defined in a separate file; they might want to define it locally and then import apps. Here are some alternatives:

- Payload could accept a string (the filename to load) or a dict (a locally-defined config)
- We could split this into two methods (naming suggestions welcome, maybe `load_file` and `load_config`?)

Reply:
@yadudoc: agreed, I've switched to just loading the config dict and rebased onto master.